huffman/test/bit_span_test.cpp

```diff
@@ -84,12 +83,71 @@ auto main() -> int
   using ::boost::ut::operator|;
 
   test("aborts if bit offset too large") = [](auto bit_offset) {
-    static constexpr auto data =
-        std::array{std::byte{0b10101010}, std::byte{0xff}};
+    static constexpr auto data = huffman::byte_array(0b10101010, 0xff);
 
     expect(aborts([&] {
       static constexpr auto bit_size = 7;
       huffman::bit_span{data.begin(), bit_size, bit_offset};
     }));
   } | std::vector<std::uint8_t>{8, 9, 10};  // NOLINT(readability-magic-numbers)
+
+  std::vector<unsigned> n_to_consume(1Z + 2Z * CHAR_BIT);
+  std::iota(n_to_consume.begin(), n_to_consume.end(), 0);
+
+  test("consume") = [](auto n) {
+    static constexpr auto data = huffman::byte_array(0b10101010, 0b01010101);
+
+    static constexpr auto nth_bit = [](auto m) {
+      return huffman::bit{std::bitset<CHAR_BIT>{
+          std::to_integer<unsigned>(data[m / CHAR_BIT])}[m % CHAR_BIT]};
+    };
+
+    auto bits = huffman::bit_span{data};
+    if (n <= data.size() * CHAR_BIT) {
+      bits.consume(n);
+      expect(bits.n_consumed() == n);
+      expect(nth_bit(n) == bits[0]);
+      expect(CHAR_BIT * data.size() - n == bits.size());
+    } else {
+      expect(aborts([&] { bits.consume(n); }));
+    }
+  } | n_to_consume;
+
+  test("consume_to_byte_boundary") = [] {
+    static constexpr auto data = huffman::byte_array(0b10101010, 0b01010101);
+    huffman::bit_span span{data};
+    expect(*span.begin() == 0_b);
```

Comment:
```suggestion
    expect(span[0] == 0_b);
```
here and below

actually `span.front()` is probably better everywhere in this file